src/tests/e2e/login.post.test.ts

```diff
@@ -0,0 +1,61 @@
+import 'dotenv/config';
+import request from 'supertest';
+import app from '../../app';
+import { mockPassword, mockUsername, mockBasicAuthString } from '../mockObjects/mockUser';
+import { setupE2E } from '../setupTests';
+import { ReasonPhrases, StatusCodes } from 'http-status-codes';
+import { userService } from '../../services/userService';
+import { mockComparePassword, mockGenerateTokens, mockGetUser } from '../mockObjects/mockUserService';
+
+setupE2E('login-e2e', () => {
+  const generateTokensSpy = jest.spyOn(userService, 'generateTokens').mockImplementation(mockGenerateTokens);
+  const getUsersSpy = jest.spyOn(userService, 'getUser').mockImplementation(mockGetUser);
+  const comparePasswordSpy = jest.spyOn(userService, 'comparePassword').mockImplementation(mockComparePassword);
+  describe('POST /login responds', () => {
+    test('with set of tokens for a logged in user', async () => {
+      const res = await request(app)
+        .post('/login')
```

Comment:
you could transfer this `/login` string to `const ENDPOINT = '/login'` and change all `.post('/login')` to `.post(ENDPOINT)`